snuba/datasets/storages/transactions.py

```diff
@@ -232,6 +233,7 @@ def transactions_migrations(
         TransactionColumnProcessor(),
         ArrayJoinKeyValueOptimizer("tags"),
         PrewhereProcessor(),
+        transform_user_to_nullable(),
```

Comment:
This seems a discover specific problem. A column is formatted in a way in a storage and in a different way in another. I think the right place for this kind of transformation is the query translator. There is where we tend to add the mapping logic between the logical and phyisical query.

Hre specifically: https://github.com/getsentry/snuba/blob/master/snuba/datasets/transactions.py#L33-L63

that is picked up by the discover logic as well.